scripts/pre_commit_linter.py

```diff
@@ -0,0 +1,151 @@
+
+"""Pre-commit script for Oppia.
+
+This script uses the JSCS node module to lint JavaScript code
+It returns and autofix errors
+
+IMPORTANT NOTES:
+
+1.  Before running this script, you must install third-party dependencies by
+    running
+
+        bash scripts/start.sh
+
+    at least once.
+
+2.  This script should be run from the oppia root folder:
+
+        python scripts/pre_commit_linter.py
+
+ Note that the root folder MUST be named 'oppia'.
+ """
+
+__author__ = "Barnabas Makonda(barnabasmakonda@gmail.com)"
+
+import argparse
+import os
+import subprocess
+import sys
+import time
+
+
+_PARSER = argparse.ArgumentParser()
+_PARSER.add_argument(
+    '--autofix',
+    help='optional; if specified, autofix and display errors which can not be fixed',
+    action='store_true')
+
+
+def _get_changed_filenames():
+    """Returns a list of modified files (both staged and unstaged)
+    Returns:
+        a list of files about to be commited.
+    """
+    unstaged_files = subprocess.check_output([
+        'git', 'diff', '--name-only']).splitlines()
+    staged_files  = subprocess.check_output([
+        'git', 'diff', '--cached', '--name-only', 
+        '--diff-filter=ACM']).splitlines()
+    return unstaged_files + staged_files
+
+
+def _is_javascript_file(filename):
+    """Check if the input filename represents a JavaScript file.
+
+    Args:
+    - filename: str. The name of the file to be checked.
+
+    Returns:
+      bool: True if the filename ends in ".js", and false otherwise.
+    """
+    return filename.endswith('.js')
+
+
+def _lint_js_files(jscs_path, autofix = None):
+    """Prints a list of lint errors in changed JavaScript files.
+
+    Args:
+    - jscs_path: path to the JSCS binary.
+    - autofix: (Optional[str]): Argument to autofix errors. Defaults to None. 
+    """
+    start_time = time.time()
+    # List of errors.
+    errors = []
+    changed_filenames = _get_changed_filenames()
+
+    # Find all javascript files.
+    changed_js_filenames = filter(
+        _is_javascript_file, changed_filenames)
+
+    # Do nothing if  no Javascript files changed.
+    if len(changed_js_filenames) == 0:
+        print 'No Javascript file to check'
+        sys.exit(0)
+
+    num_js_files = len(changed_js_filenames)
+
+    for ind, filename in enumerate(changed_js_filenames):
+        print 'Linting %s (file %d/%d)...\t' % (
+            filename, ind + 1, num_js_files)
+
+        proc_args = (
+            [jscs_path, '-x', filename] if autofix
+            else [jscs_path, filename])
+        try:
+            proc = subprocess.Popen(
+                proc_args,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE)
+            linter_stdout, linter_stderr = proc.communicate()
+            errors.append(linter_stdout)
+            print linter_stdout
+        except OSError as e:
+            print e
+            sys.exit(1)
+
+    # This removes empty string added in case no errors.
```

Comment:
Here is a little bit tricky I think we might better use true or false because actually the output errors for a given file is just one long string and I tried to look and we can not really show how many errors the file has 
![console](https://cloud.githubusercontent.com/assets/6409210/10557031/820d4776-74a2-11e5-9dfb-8c4c4c55e46e.png)
like the above screenshot, all that summary is outputted as one long string. 
 but jscs itself give summary of error bellow the printed errors.
so even if we increment num_errors we will end up getting just number of js files.